plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDAssociation.java

```diff
@@ -68,25 +68,27 @@ public ERDAssociation(ERDElement<?> sourceEntity, ERDElement<?> targetEntity, bo
 
     /**
      * Constructor for physical association
-     * @param association physical FK
+     *
+     * @param association  physical FK
      * @param sourceEntity fk table
      * @param targetEntity pk table
-     * @param reflect reflect flag
+     * @param reflect      reflect flag
      */
-	public ERDAssociation(DBSEntityAssociation association, ERDEntity sourceEntity, ERDEntity targetEntity, boolean reflect)
-	{
-		super(association);
-		this.targetEntity = targetEntity;
-		this.sourceEntity = sourceEntity;
-
-		// Resolve association attributes
+    public ERDAssociation(
+        DBSEntityAssociation association,
+        @NotNull ERDEntity sourceEntity,
+        @NotNull ERDEntity targetEntity,
+        boolean reflect) {
```

Comment:
```suggestion
    public ERDAssociation(
        DBSEntityAssociation association,
        @NotNull ERDEntity sourceEntity,
        @NotNull ERDEntity targetEntity,
        boolean reflect
    ) {
```

Reply:
line 82 looks very strange for me.

Reply:
but that's how we do and we discussed it several times already